test/EVM/Test/FuzzSymExec.hs

```diff
@@ -760,38 +761,39 @@ checkTraceAndOutputs contract gasLimit txData = do
   case hevmRun of
     (Right (expr, hevmTrace, hevmTraceResult)) -> liftIO $ do
       let
-        concretize :: Expr a -> Expr Buf -> Expr a
-        concretize a c = mapExpr go a
+        concretize :: Expr Buf -> Expr a ->  Expr a
+        concretize c a = mapExpr go a
           where
             go :: Expr a -> Expr a
             go = \case
                        AbstractBuf "calldata" -> c
                        y -> y
-        concretizedExpr = concretize expr (ConcreteBuf txData)
-        simplConcExpr = Expr.simplify concretizedExpr
+        concretizedExpr = map (concretize (ConcreteBuf txData)) $ expr
+        simplConcExpr = map Expr.simplify concretizedExpr
         getReturnVal :: Expr End -> Maybe ByteString
         getReturnVal (Success _ _ (ConcreteBuf bs) _) = Just bs
         getReturnVal _ = Nothing
-        simplConcrExprRetval = getReturnVal simplConcExpr
+        simplConcrExprRetval = mapMaybe getReturnVal simplConcExpr
       traceOK <- compareTraces hevmTrace (evmtoolTraceOutput.trace)
       -- putStrLn $ "HEVM trace   : " <> show hevmTrace
       -- putStrLn $ "evmtool trace: " <> show (evmtoolTraceOutput.trace)
       assertEqual "Traces and gas must match" traceOK True
       let resultOK = evmtoolTraceOutput.output.output == hevmTraceResult.out
       if resultOK then liftIO $ do
         putStrLn $ "HEVM & evmtool's outputs match: '" <> (bsToHex $ bssToBs evmtoolTraceOutput.output.output) <> "'"
-        if isNothing simplConcrExprRetval || (fromJust simplConcrExprRetval) == (bssToBs hevmTraceResult.out)
+        assertBool "Cannot have more than one success return value" (length simplConcrExprRetval <= 1)
+        if null simplConcrExprRetval || (simplConcrExprRetval !! 0) == (bssToBs hevmTraceResult.out)
            then do
              putStr "OK, symbolic interpretation -> concrete calldata -> Expr.simplify gives the same answer."
-             if isNothing simplConcrExprRetval then putStrLn ", but it was a Nothing, so not strong equivalence"
-                                               else putStrLn ""
+             if null simplConcrExprRetval then putStrLn ", but it was a Nothing, so not strong equivalence"
+                                          else putStrLn ""
            else do
              putStrLn $ "original expr                    : " <> (show expr)
              putStrLn $ "concretized expr                 : " <> (show concretizedExpr)
              putStrLn $ "simplified concretized expr      : " <> (show simplConcExpr)
              putStrLn $ "evmtoolTraceOutput.output.output : " <> (show (evmtoolTraceOutput.output.output))
              putStrLn $ "HEVM trace result output         : " <> (bsToHex (bssToBs hevmTraceResult.out))
-             putStrLn $ "ret value computed via symb+conc : " <> (bsToHex (fromJust simplConcrExprRetval))
+             putStrLn $ "ret value computed via symb+conc : " <> (intercalate "," $ map bsToHex simplConcrExprRetval)
```

Comment:
Seems unnecessary to use `intercalate` here because you know `simplConcrExprRetval` has exactly one element.

I think something like this should work?
```suggestion
             putStrLn $ "ret value computed via symb+conc : " <> (bsToHex (simplConcrExprRetval !! 0))
```